### PR TITLE
Add option to use FASTA files with protein sequences as input

### PR DESCRIPTION
**Pull request by skrakau**
Since I didn't want to write python2 scripts, I added a container for python3 ;) would that be OK?
The FASTA file is processed and peptides with the chosen lengths are generated from all sequences. Unique peptides together with a list of proteins in which they occur as well as a list with corresponding counts are stored and used for epitope prediction. For now only one FASTA file is supported as input.

I called the input parameter `--proteins`. Maybe we want to change the name of the `--proteome` parameter (used for self-filtering), since this might be a bit confusing?

#
Many thanks for contributing to nf-core/epitopeprediction!

Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

## PR checklist
 - [ ] This comment contains a description of changes (with reason)
 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [x] If necessary, also make a PR on the [nf-core/epitopeprediction branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/epitopeprediction)
 - [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [ ] Make sure your code lints (`nf-core lint .`).
 - [ ] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/epitopeprediction/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: bin/gen_peptides.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+parser = argparse.ArgumentParser("Preprocessing of epitopes.")
+parser.add_argument('-i', '--input', metavar='FILE', type=str, help = 'TSV filename, containing paths to fasta files.')
+parser.add_argument('-o', '--output', metavar='FILE', type=argparse.FileType('w'), help='Output file: peptides.')
+parser.add_argument('-min', '--min_length', metavar='N', type=int, help='Min. length of peptides that will be generated.')
+parser.add_argument('-max', '--max_length', metavar='N', type=int, help='Max. length of peptides that will be generated.')
+args = parser.parse_args()
+
+
+# for each fasta file get proteins: Id and sequence
```
Comment (christopher-mohr) [inline bin/gen_peptides.py]: Maybe we should use here the functionality already provided by FRED2: https://github.com/FRED-2/Fred2/blob/master/Fred2/IO/FileReader.py

File: conf/base.config [hunk 2]
```diff
@@ -42,6 +42,9 @@ process {
   withLabel:process_long {
     time = { check_max( 20.h * task.attempt, 'time' ) }
   }
+  withName:genPeptides {
+    container = 'nfcore/epitopeprediction-py3:dev'
```
Comment (christopher-mohr) [inline conf/base.config]: I'm don't know if that works. Should probably be reviewed by someone else.

File: containers/py3/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
+FROM nfcore/base:1.9
+LABEL authors="Sabrina Krakau" \
+      description="Docker image containing python3 for gen_peptides.py in the  nf-core/epitopeprediction pipeline"
+
```
Comment (christopher-mohr) [inline containers/py3/Dockerfile]: See previous comment about python3 container

File: conf/test_proteins.config [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ *   nextflow run nf-core/epitopeprediction -profile test_proteins
+ */
+
+params {
+  max_cpus = 2
+  max_memory = 6.GB
+  max_time = 4.h
+
+  // Input data
+  peptides = 'https://github.com/nf-core/test-datasets/raw/epitopeprediction/testdata/proteins/proteins.tsv'
```
Comment (christopher-mohr) [inline conf/test_proteins.config]: Shouldn't this be ```proteins = ...``` ? As far as I understand this is about testing the protein FASTA functionality?

File: containers/py3/environment.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# You can use this file to create a conda environment for this pipeline:
+#   conda env create -f environment.yml
+name: nf-core-epitopeprediction-py3-1.1.0dev
```
Comment (christopher-mohr) [inline containers/py3/environment.yml]: See comment about python3 above

File: bin/gen_peptides.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+protid_protseq = pd.DataFrame(
+    [ (prot_id, prot_seq) for prot_id, prot_seq in get_proteins(args.input)  ],
+    columns = ['protein','sequence']
+    )
+
+
+####################
+# generate peptides
+def gen_peptides(prot_seq, k):
+    return [ prot_seq[i:(i+k)] for i in range(len(prot_seq)-k) ]
```
Comment (christopher-mohr) [inline bin/gen_peptides.py]: same as above, we should use the FRED2 functionality to generate peptides from proteins in my opinion

File: bin/gen_peptides.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+# for each protein sequence, for each length k: generate all peptides
```
Comment (christopher-mohr) [inline bin/gen_peptides.py]: I guess we can simply this a lot by using FRED2 functionality. Besides, as discussed yesterday I see that it makes sense to count occurrences for your project. But I'm wondering if we should include it in general?

Comment (skrakau) [reply bin/gen_peptides.py]: I think the counts should be included, at least optionally?

File: conf/test_proteins.config (hunk 4 above)
Comment (skrakau) [reply conf/test_proteins.config]: ups, thanks!

**Comment by skrakau** [comment]
I think the counts might be important for other applications as well. But FRED2s `generate_peptides_from_proteins` also provides information about the positions, so the information about the counts is there. I just got rid of FRED2 for my personal scripts, because of the python2 problem, but I agree it would be much more consistent.

File: bin/gen_peptides.py (hunk 7 above)
Comment (christopher-mohr) [reply bin/gen_peptides.py]: In this case I would just include it in order to avoid introducing another parameter.

**Comment by skrakau** [comment]
see #42

**Pull request closed without merging**